test/splitOnRegex.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var jsc = require('jsverify');
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+var equals = require('./internal/equals');
+
+
+test('splitOnRegex', function() {
+
+  eq(typeof S.splitOnRegex, 'function');
+  eq(S.splitOnRegex.length, 2);
+  eq(S.splitOnRegex.toString(), 'splitOnRegex :: RegExp -> String -> Array String');
+
+  eq(S.splitOnRegex(/b/, 'abc'), ['a', 'c']);
+  eq(S.splitOnRegex(/d/, 'abc'), ['abc']);
+  eq(S.splitOnRegex(/b(?=c)/, 'abc abd'), ['a', 'c abd']);
+  eq(S.splitOnRegex(/b(?!c)/, 'abc abd'), ['abc a', 'd']);
+
+  eq(S.splitOnRegex(/\s+/, 'foo  bar baz   qux'), ['foo', 'bar', 'baz', 'qux']);
+  eq(S.splitOnRegex(/\s+/, 'foo  bar baz\tqux\nquux'), ['foo', 'bar', 'baz', 'qux', 'quux']);
+  eq(S.splitOnRegex(/\s+/, 'foobar'), ['foobar']);
+  eq(S.splitOnRegex(/;(;)?/, 'foo;;bar;baz'), ['foo', 'bar', 'baz']);
```

Comment:
I like the fact that we're testing a capturing group, but it bothers me slightly that a capturing group is not required in this case. I'm struggling to think of a good example, but we could use something like this:

```javascript
eq(S.splitOnRegex(/([:;])\1/, 'foo::bar:;baz;;quux'), ['foo', 'bar:;baz', 'quux']);
```